app.rb

```diff
@@ -1,20 +1,326 @@
+
+
+
+# class Property_mangement
+
+#   def initialize()
+#   end
+
+
+
+# # First, Open the data.rb an inspect the data. Identify and write, in comments, the following:
+#   # Explain how the data is structured
+#   # The data is structured as two tables.  Each table with fields pertinent to the table name of Apatments ad Tenants.  The tables are linked by the apartment id existing as a foreign key in the Tenants table.
+#   # What are the properties for each of the two types of hashes
+#   # The properties for each hash are: For the Apartments hash; id; address; rent; square feet.
+#   # For the Tenant hash: id; name; age; apartment id.
+
+# # Use enumerables to -
+#   # Print all the addresses for the apartments
+
+#   def print_all_addresses
+#     @apartments.map do |add|
+#       the_address = add[:address]
+#       puts the_address
+#     end
+#   end
+
+
+#   def print_all_tenant_names
+#     @tenants.map do |name|
+#       the_name = name[:name]
+#       puts the_name
+#     end
+#   end
+
+#   def print_cheap_apts
+#     puts @apartments.find_all {|apt| apt[:monthly_rent] < 700} 
+#   end
+
+#   def print_tenants_over_age
+#     puts @tenants.find_all {|peeps| peeps[:age] > 44}
+#   end
+
+
+#   def print_tenants_in_apt
+#     puts @tenants.find_all {|peeps| peeps[:apartment_id] == 1}
+#   end
+
+#   def print_tenants_by_age
+#     sorted_tenants = @tenants.sort_by do |item|
+#       item[:age]
+#     end
+#     puts sorted_tenants
+#   end
+
+
+#   def print_tenants_alpha
+#     sorted_tenants = @tenants.sort_by do |item|
+#       item[:name]
+#     end
+#     puts sorted_tenants
+#   end
+
+#   # More challenging
+#   # When printing tenants also print out the address that the tenant resides in.
+
+#   def print_tenant_and_apt
+#     @tenants.each do |item|
+#       @tenants_apartment_id = item[:apartment_id]
+#       puts item
+#       puts @apartments[@tenants_apartment_id]
+#       puts "*" * 50
+#     end
+#   end
+# end
+
+
+  # When printing all apartments, under each apartment print all of its tenants
+
+
+# bob = Property_mangement.new
+
+
 require_relative "data"
 apartments = data[:apartments]
 tenants = data[:tenants]
+choice = ""
+
+
+
+class Apartment
+  attr_accessor(:apt_id, :addr, :rent, :size)
+  def initialize(id, addr, rent, size)
+    @apt_id = id
+    @addr = addr
+    @rent = rent
+    @size = size
+  end
+  #   # Print all the addresses for the apartments
+
+  def print_all_addresses
```

Comment:
the problem with this method is that we're using a variable we can't access `apartments`. Now that we've seen active record we have methods like `.all` that queries our database and actually creates new instances for every row returned.